### DISCUSSION TITLE
Streaming vs PPR

### DISCUSSION THREAD

**Discussion by Gallevy** [remix-run/remix#9367] [Q&A] [answered]
Is streaming with `defer` and `Await` equivalent to Next PPR.
I just recently got into remix and still trying to learn the basic and i couldn't figure out if they are working the same or there is some differences I don't understand,

Thanks

**Comment by sergiodxa** [comment, marked as answer]
Not exactly, Next PPR is an evolution of SSG, where the static apps (usually an app shell) is generated as HTML at build and then when a request comes the hosting can start sending that static HTML and stream the dynamic data.

While `Await` is like `React.use` new hook coming in v19 (Await will most likely be deprecated once v19 is stable), and `defer` that's being deprecated now with Single Fetch allows you to as the name says defer data, usually this is done for non critical data.

This means that even with defer the browser will wait for all loaders to finish getting any critical (not deferred) data and then it will get a response.

The deferred data will be streamed to the browser the same way Next does for the dynamic data, but here there's no app shell statically generated.
